applications/GeoMechanicsApplication/custom_processes/apply_k0_procedure_process.cpp

```diff
@@ -43,37 +43,65 @@ void SetConsiderDiagonalEntriesOnlyAndNoShear(ModelPart::ElementsContainerType&
 namespace Kratos
 {
 
-ApplyK0ProcedureProcess::ApplyK0ProcedureProcess(ModelPart& model_part, Parameters K0Settings)
-    : Process(Flags()), mrModelPart(model_part), mSettings(std::move(K0Settings))
+ApplyK0ProcedureProcess::ApplyK0ProcedureProcess(Model& rModel, const Parameters& K0Settings)
+    : mSettings(std::move(K0Settings))
 {
+    KRATOS_ERROR_IF_NOT(mSettings.Has("model_part_name") || mSettings.Has("model_part_name_list"))
+        << "Please specify 'model_part_name' or 'model_part_name_list' for "
+        << ApplyK0ProcedureProcess::Info();
+
+    KRATOS_ERROR_IF(mSettings.Has("model_part_name") && mSettings.Has("model_part_name_list"))
+        << "The parameters 'model_part_name' and 'model_part_name_list' are mutually exclusive for "
+        << ApplyK0ProcedureProcess::Info();
+
+    if (mSettings.Has("model_part_name_list")) {
+        const auto model_part_names = mSettings["model_part_name_list"].GetStringArray();
+        KRATOS_ERROR_IF(model_part_names.empty()) << "The parameters 'model_part_name_list' needs "
+                                                     "to contain at least one model part name for "
+                                                  << ApplyK0ProcedureProcess::Info();
+        std::ranges::transform(
+            model_part_names, std::back_inserter(mrModelParts),
+            [&rModel](const auto& rName) -> ModelPart& { return rModel.GetModelPart(rName); });
+    } else {
+        mrModelParts = {rModel.GetModelPart(mSettings["model_part_name"].GetString())};
+    }
```

Comment:
I would probably always construct a list of names (which, in the case of `"model_part_name"` would just contain a single value). In that way, how to get a model part by name from the model needs to be written down only once, not twice:
```suggestion
    const auto model_part_names = mSettings.Has("model_part_name_list") ? mSettings["model_part_name_list"].GetStringArray() : std::vector<std::string>{mSettings["model_part_name"].GetString()};
    KRATOS_ERROR_IF(model_part_names.empty()) << "The parameters 'model_part_name_list' needs "
                                                 "to contain at least one model part name for "
                                              << ApplyK0ProcedureProcess::Info();
    std::ranges::transform(
        model_part_names, std::back_inserter(mrModelParts),
        [&rModel](const auto& rName) -> ModelPart& { return rModel.GetModelPart(rName); });
```

Reply:
Nice suggestion indeed, done!